packages/core/src/events/augury-event.class.ts

```diff
@@ -1,29 +1,41 @@
-import { AuguryEventName } from './augury-event-name.type';
+import { Probe } from '../probes';
+import { DragPeriod } from './drag-period.class';
 
 export class AuguryEvent {
   private static NextId = 0;
 
   public readonly id: number;
-  public readonly payload: any;
-  public readonly creationAtTimestamp: number;
-  public readonly completedAtTimestamp: number;
-
-  constructor(
-    public readonly probeName: string,
-    public readonly name: AuguryEventName,
-    createPayload: () => any,
-  ) {
+  public readonly probeName: string;
+  private timestamps: DragPeriod = null;
+
+  constructor(probe: Probe) {
     this.id = AuguryEvent.NextId++;
-    this.creationAtTimestamp = performance.now();
-    this.payload = createPayload();
-    this.completedAtTimestamp = performance.now();
+    this.probeName = probe.constructor.name;
+  }
+
+  public set dragPeriod(dragPeriod: DragPeriod) {
+    if (this.timestamps) {
+      throw new Error('The drag period is already set!');
+    }
+
+    dragPeriod.markComplete();
+
+    this.timestamps = dragPeriod;
+  }
+
+  public get dragPeriod(): DragPeriod {
+    return this.timestamps;
+  }
+
+  public isInstanceOf(clazz: any): boolean {
```

Comment:
or `klass` ;) Not sure if it would be helpful to create a type alias to infer the intention of the any (e.g. `type EventClassType = any` & `clazz: EventClassType`)

Reply:
I'm indifference to aClass, clazz or klass lol. I've seen them all.

Reply:
I really wish typescript had a Class class like Java does. Super annoying :(